### PR TITLE
1159 check functional

### LINKED ISSUE #1159: Enable librina and rinad tests as part of the continuous integration build process
Given that the [IRATI Stack now builds on the Travis CI](https://github.com/IRATI/stack/issues/1109) service the next step is to enable the librina and rinad tests to see how they support and assist the continuous development and deployment of the Stack.

### PR DESCRIPTION
**Pull request by miguelpdl**
Closes #1159 

Merged all the latest changes with the arcfire branch onto this branch and added to Travis CI and Gitlab Runner builds to make sure that the make check tests are run in librina and rinad.

The build passes on both the Travis CI build as well as the GitLab Runner so this should be good to go into the arcfire branch now.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,7 +35,26 @@ of IRATI in future releases.
...
+[![Build Status of the Master branch on Ubuntu](https://travis-ci.org/IRATI/stack.svg?branch=master)](https://travis-ci.org/IRATI/stack)
+
+Build Status of arcfire branch on Ubuntu via Travis CI.
+
+[![Build Status of arcfire branch on Ubuntu](https://travis-ci.org/IRATI/stack.svg?branch=arcfire)](https://travis-ci.org/IRATI/stack)
+
+
+
+### Building on Ubuntu 16.04 and Debian 8, Debian 9
+#############################################################################
```
Comment (vmaffione) [inline README.md]: Please remove this line, as it does not render in a nice way.

File: ci/build-all [hunk 2]
```diff
@@ -19,3 +19,9 @@ mkdir build
 mkdir -p $INSTALL_PREFIX
 ./configure --prefix $INSTALL_PREFIX
 sudo make install
+cd build/librina
+sudo make check-functional
+cd ../..
+sudo make install
```
Comment (vmaffione) [inline ci/build-all]: why repeating `sudo make install`?

File: ci/build-all [hunk 3]
```diff
@@ -19,3 +19,9 @@ mkdir build
 mkdir -p $INSTALL_PREFIX
 ./configure --prefix $INSTALL_PREFIX
 sudo make install
+cd build/librina
+sudo make check-functional
```
Comment (vmaffione) [inline ci/build-all]: Can we get rid of `sudo`?

File: ci/build-all [hunk 4]
```diff
@@ -19,3 +19,9 @@ mkdir build
...
+cd ../..
+sudo make install
+cd build/rinad
+sudo make check
```
Comment (vmaffione) [inline ci/build-all]: Can we get rid of `sudo`?

File: README.md (hunk 1 above)
Comment (miguelpdl) [reply README.md]: Ahh yes too true, fixed that up now.

File: ci/build-all (hunk 2 above)
Comment (miguelpdl) [reply ci/build-all]: That is a hang over from the previous Travis CI way of building this. It is now removed with the last commit.

File: ci/build-all (hunk 3 above)
Comment (miguelpdl) [reply ci/build-all]: Yes removed sudo now

File: ci/build-all (hunk 4 above)
Comment (miguelpdl) [reply ci/build-all]: Yes removed sudo now

File: ci/build-all [hunk 5]
```diff
@@ -18,4 +18,9 @@ sudo ldconfig
 mkdir build
 mkdir -p $INSTALL_PREFIX
 ./configure --prefix $INSTALL_PREFIX
-sudo make install
+make install
```
Comment (vmaffione) [inline ci/build-all]: I guess you mistakenly removed sudo here (where it is needed), while you wanted to remove it from `make check-functional` below :)

Comment (miguelpdl) [reply ci/build-all]: Ahh crap yes, replaced now, and will watch out on the Travis CI build service now.

**Comment by edugrasa** [comment]
Make check in rinad failed when building the test classes, probably that sudo was required? (I know it shouldn't be required, the build process needs to be rethought and simplified..)

**Comment by miguelpdl** [comment]
@edugrasa yes the `sudo make check` was required as it is trying to access some directory file called .libs/test-routing. 

Error output is
```
cannot open output file .libs/test-routing: Permission denied
```

So shall we wait to have this fixed or accept the current state of affairs. The build and tests pass with `sudo make check` in place.

**Comment by vmaffione** [comment]
@miguelpdl , let's just add `sudo`, as you did.
My question "can we get rid of sudo?" was just a question, I was not assuming we could :)
Without sudo the travis CI build failed, while now it's ok.

**Pull request merged**